hikari/impl/entity_factory.py

```diff
@@ -3744,3 +3745,81 @@ def deserialize_sku(self, payload: data_binding.JSONObject) -> monetization_mode
             slug=payload["slug"],
             flags=monetization_models.SKUFlags(payload["flags"]),
         )
+
+    ###############
+    # POLL MODELS #
+    ###############
+    def _deserialize_poll_media(self, payload: data_binding.JSONObject) -> poll_models.PollMedia:
+        emoji_payload = payload.get("emoji")
+        return poll_models.PollMedia(
+            text=payload.get("text"), emoji=self.deserialize_emoji(emoji_payload) if emoji_payload else None
+        )
+
+    def _deserialize_poll_answer_count(self, payload: data_binding.JSONObject) -> poll_models.PollAnswerCount:
+        return poll_models.PollAnswerCount(
+            answer_id=payload["answer_id"], count=payload["count"], me_voted=payload["me_voted"]
+        )
+
+    def deserialize_poll(self, payload: data_binding.JSONObject) -> poll_models.PollObject:
+        question = payload["question"]
+        expiry = time.iso8601_datetime_string_to_datetime(payload["expiry"])
+        allow_multiselect = payload["allow_multiple_options"]
+        layout_type = poll_models.PollLayoutType(payload["layout_type"])
+
+        answers: typing.MutableMapping[int, poll_models.PollAnswer] = {}
+        for _answer_payload in payload["answers"]:
+            answer_id = _answer_payload["answer_id"]
+            poll_media = self._deserialize_poll_media(_answer_payload)
+
+            answers[answer_id] = poll_models.PollAnswer(answer_id=answer_id, poll_media=poll_media)
+
+        _result_payload: typing.Optional[data_binding.JSONObject] = None
+        if (_result_payload := payload.get("result")) is not None:
+            is_finalized = _result_payload.get("is_finalized")
+
+            answer_counts = tuple(
+                self._deserialize_poll_answer_count(item) for item in _result_payload.get("answer_counts")
+            )
+            results = poll_models.PollResult(is_finalized=is_finalized, answer_counts=answer_counts)
+        else:
+            results = None
```

Comment:
```suggestion
        results = None
        if (result_payload := payload.get("result")) is not None:
            is_finalized = result_payload["is_finalized")

            answer_counts = tuple(
                self._deserialize_poll_answer_count(item) for item in _result_payload["answer_counts"]
            )
            results = poll_models.PollResult(is_finalized=is_finalized, answer_counts=answer_counts)
```
The "get" i replaced because otherwise you will just get nasty `cannot access stuff from None` errors